app/models/breakeven_table_question.rb

```diff
@@ -0,0 +1,41 @@
+class BreakevenTableQuestion
+  def initialize(breakeven_data)
+    @breakeven_data = JSON.parse(breakeven_data) if breakeven_data.present?
+  end
+
+  def report
+    report = {
+      revenue: [],
+      total_revenue: 0,
+      cogs: [],
+      total_cogs: 0,
+      gross_margin: 0,
+      fixed_costs: [],
+      total_fixed_costs: 0,
+      net_margin: 0,
+    }
+
+    @breakeven_data['products'].each do |product|
+      report[:revenue] << {
+        name: product['name'],
+        quantity: product['quantity'],
+        amount: product['price'],
+        total: product['price'] * product['quantity'],
+      }
+      report[:cogs] << {
+        name: product['name'],
+        quantity: product['quantity'],
+        amount: product['cost'],
+        total: product['cost'] * product['quantity'],
+      }
+    end
+    report[:total_revenue] = report[:revenue].map { |i| i[:total] }.sum
+    report[:total_cogs] = report[:cogs].map { |i| i[:total] }.sum
+    report[:gross_margin] = report[:total_revenue] - report[:total_cogs]
+    report[:fixed_costs] = @breakeven_data['fixed_costs'].map(&:symbolize_keys)
+    report[:total_fixed_costs] = report[:fixed_costs].map { |i| i[:amount] }.sum
+    report[:net_margin] = report[:gross_margin] - report[:total_fixed_costs]
```

Comment:
These calculations (there are a lot of them) all seem to assume no nil values. I think this is a problem because we don't currently have any validation on the questionnaire. So you will get errors if you try to multiply something by nil, for instance.

I think it might be better to check each revenue and cogs line and if price/quantity/cost is nil then simply ignore them. Thoughts?